### PR TITLE
fix: refine fee breakdown errors and enforce filled status

### PR DESCRIPTION
**Pull request by dohaki**
Not 100% sure if this will fix ACX-1741 but should at least give a bit more insight in the logs I hope.

### CODE REVIEW

File: src/modules/scraper/adapter/messaging/FeeBreakdownConsumer.ts [hunk 1]
```diff
@@ -31,8 +31,16 @@ export class FeeBreakdownConsumer {
...
 
-    if (!deposit.token || !deposit.price || !deposit.depositDate) {
-      throw new Error("Can not get fee breakdown without token, price or deposit date");
+    if (!deposit.token) {
+      throw new Error("Token not populated");
+    }
+
+    if (!deposit.price) {
+      throw new Error("Price not populated");
+    }
```
Comment (james-a-morris) [inline src/modules/scraper/adapter/messaging/FeeBreakdownConsumer.ts:40]: Do we have access to the SDK? If so, can we use `isDefined()`

**Review by james-a-morris — COMMENTED** [review]
Left a comment

File: src/modules/scraper/adapter/gas-fees/gas-fees-service.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class GasFeesService {
     const fillTxGasCostWei = new BigNumber(fillTxReceipt.effectiveGasPrice).multipliedBy(fillTxReceipt.gasUsed);
     const fillTxBlock = await this.providers.getCachedBlock(destinationChainId, fillTxReceipt.blockNumber);
     const nativeTokenPriceUsd = await this.marketPriceService.getCachedHistoricMarketPrice(
-      fillTxBlock.date,
+      new Date(fillTxBlock.date),
```
Comment (amateima) [inline src/modules/scraper/adapter/gas-fees/gas-fees-service.ts:24]: `fillTxBlock.date` is already a Date 🤔

Comment (dohaki) [reply src/modules/scraper/adapter/gas-fees/gas-fees-service.ts]: oh yea true. reverted this change

File: src/modules/scraper/adapter/messaging/FeeBreakdownConsumer.ts (hunk 1 above)
Comment (dohaki) [reply src/modules/scraper/adapter/messaging/FeeBreakdownConsumer.ts:40]: No, we currently don't use the SDK the Scraper. If we need it at some point, we could use the helper yea

**Review by amateima — APPROVED** [review]

**Pull request merged**